### ISSUE TITLE
Random hangs

### ISSUE THREAD

**Issue by lotsofs** [LiveSplit/LiveSplit#2385]
LiveSplit has recently (over the last year) started hanging at random, like it gets stuck in an infinite loop or something. This typically happens after a few hours or half a day of running, but sometimes happens mere minutes after launching the software. It happens both when the timer is running, and when it is not.

OS: Windows 10

Symptoms:
* The timer stops, autosplitter also seems to not be doing its checks anymore
* The window can no longer be dragged around, and the cursor changes to the ↔ arrow when hovering over it.
* If I right click the taskbar icon and select Quit Program, it will prompt me whether I want to save my splits as normal. Selecting yes or no appears to shut down the program. However, the livesplit.exe process persists in the background and can be found in the task manager. I have yet to determine whether selecting 'yes' will actually save my splits or not.
* No logs appear in the event viewer.
* I've done a Process Explorer thing by looking at the Stack of the Livesplit.exe thread. I do not know how to make sense of it, but maybe it helps:

![image](https://github.com/LiveSplit/LiveSplit/assets/18078548/6fb1d234-bcd2-4524-a360-244641b40deb)

![image](https://github.com/LiveSplit/LiveSplit/assets/18078548/627f21ef-5f5f-4e16-bcf1-de48c2314b4c)

I suspect it has something to do with running the Scriptable Autosplitter component in the layout as this started happening after I decided to add it to my layout. I will run tests with and without the component over the next couple nights.

Please guide me as to what other next steps I can take to debug this issue.

**Comment by wooferzfg** [comment]
It could be an issue with the autosplitter you're using?

**Comment by KR33PYK1NG** [comment]
Can confirm I've also had issues with random freezing for a while. LiveSplit just stops responding and I have to kill the process.

**Comment by lotsofs** [comment]
Ran a test with the default layout and no autosplitter and it still froze, about 16 hours in this time. It appears that autosplitters are not a factor, and the first freeze I had was just coincidence.
![image](https://github.com/LiveSplit/LiveSplit/assets/18078548/c73e408b-28f9-4a63-acba-dc903e26a91c)

Earlier I ran a test of three livesplits running at once, and they all froze at exactly the same time, about 19 hours in.

I've had several freezes in the meantime with the timer not running, freezing at a couple hours in (I didn't measure)

**Comment by wooferzfg** [comment]
Can you confirm that "Allow Gamepads as Hotkeys" is disabled in the settings, and that you do not have "LiveSplit.UnrealLoads.dll" in your component folder?

**Comment by wooferzfg** [comment]
@lotsofs Do you have any event logs from the time of the freeze?

1. Go to your Control Panel.
2. Search for Event Viewer and open it.
3. Click "Windows Logs".
4. Click "Application".
4. Select the last few LiveSplit errors. Make sure that they match the time when the error happened.
5. Save Selected Events (save them as the default format).
6. Send the file to me (you can add me on Discord `@wooferzfg`).

**Comment by wooferzfg** [comment]
Also, this looks like a similar issue based on the stack trace above: https://stackoverflow.com/q/8314233

@lotsofs For the stack trace you posted above, did LiveSplit freeze while you weren't using your computer?

**Comment by lotsofs** [comment]
The stack traces above were all for livesplit processes that froze while I was asleep/away but my computer was not (I simply turned the monitors off). I think so anyway, the window inevitably does end up in the background as I use my computer for other things and I will only notice it has frozen when I shut programs down at the end or start of the day. I will 

Earlier the timer froze at about 12 hours while I was doing a 14 hour speedrun on the same PC, which is the first time it ever bothered me and prompted me to file this report. (though it did freeze 5 minutes in a run a few weeks earlier, but that was just an "oh well, Ill just reset" case.) I did not gather a stack trace for either of those instances though

**Comment by KR33PYK1NG** [comment]
Completely blocking in/out net traffic for LiveSplit seems to have fixed the freezes for me (or maybe I was very lucky to not encounter them yet? Still, wanted to point this out as a possible solution).
@lotsofs Could you try blocking the traffic aswell to confirm whether it's placebo or not?

**Comment by lotsofs** [comment]
@KR33PYK1NG So far it's going 21 hours without a freeze yet, which is a new record over the past month or so. I will keep it running to see if it eventually freezes and restart the application at ~48 hours and start over to rule out flukes. If it works this would be an acceptable workaround to me, though the underlying issue obviously still persists. I guess this knowledge also helps with debugging.

**Comment by BOBPRESIDENT** [comment]
@KR33PYK1NG Was just wondering how to go about blocking in/out net traffic for Livesplit?

**Comment by lotsofs** [comment]
Can confirm my LiveSplit stopped hanging unless extreme flukes are taking place. Obviously not a perfect solution because now there will be no auto updates, automatic sharing, etc etc.

@BOBPRESIDENT I used this tutorial to block LiveSplit in my firewall both inbound and outbound: https://www.enterprisestorageforum.com/security/block-a-program-in-firewall/

**Comment by BOBPRESIDENT** [comment]
Were you ever able to get any of your splits back from frozen instances?

**Comment by lotsofs** [comment]
No, though I also didn't try that hard since neither of those runs ended up PBing. However, the freeze appears to be not just a visual bug; I don't think the software continues timing and splitting in the background (judging from what little I understand of the process monitor) nor does it actually seem to save when yes is replied to the save prompt.

**Comment by CryZe** [comment]
Thanks for figuring out that it's network related. Maybe we can pinpoint it further whether it's the speedrun.com, SRL, or racetime.gg integration. I think one of the next steps could be to disable those racing services in the settings.

**Comment by lotsofs** [comment]
I have untuck both the racetime.gg & SRL services in these settings, speedrun.com does not appear, and will now run a test with both these 2 services disabled. Will report back in 48 hours
![image](https://github.com/LiveSplit/LiveSplit/assets/18078548/07c7dd23-e79a-4886-958a-d05fcb836b80)

**Comment by wooferzfg** [comment]
@lotsofs When you were having the freezing issue, was LiveSplit showing an update prompt on startup or no?
